src/legacy/ui/public/new_platform/index.test.mocks.ts

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export const mockNewPlatformBackdoor = () => {
```

Comment:
The name and location of this file suggests that it's a generic New Platform mock so other developers might also use this in their tests although the `plugins.data` mock is very specific to your unit tests. We'd rather want them to use https://github.com/elastic/kibana/pull/39351

Can you move this mock file into your plugin so that the code is more local?

Optional: Use / adapt https://github.com/elastic/kibana/pull/39351 for your tests. You'll have to add an empty `plugins` mock to https://github.com/elastic/kibana/pull/39351/files#diff-a0d66a99cc197c9ae8a0956f8df3bbdeR28

and then in your tests:
```typescript
import { pluginsMock } from '../../../../../../../../core/public/ui_new_platform.test.mocks';
pluginsMock.mockReturnValue({
 data: {
   expressions: {...}
}    
```

Reply:
Created PR: https://github.com/elastic/kibana/pull/40123